src/include/catalog/postgres/pg_constraint_impl.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <vector>
+
+#include "common/managed_pointer.h"
+#include "storage/projected_row.h"
+#include "storage/storage_defs.h"
+
+namespace noisepage::parser {
+class AbstractExpression;
+}  // namespace noisepage::parser
+
+namespace noisepage::storage {
+class RecoveryManager;
+class SqlTable;
+
+namespace index {
+class Index;
+}  // namespace index
+}  // namespace noisepage::storage
+
+namespace noisepage::transaction {
+class TransactionContext;
+}  // namespace noisepage::transaction
+
+namespace noisepage::catalog::postgres {
+class Builder;
+
+/** The NoisePage version of pg_constraint. */
+class PgConstraintImpl {
+ public:
+  /**
+   * Prepare to create pg_constraint.
+   * Does NOT create anything until the relevant bootstrap functions are called.
+   *
+   * @param db_oid          The OID of the database that pg_constraint should be created in.
+   */
+  explicit PgConstraintImpl(db_oid_t db_oid);
+
+  /** Bootstrap the projected row initializers for pg_constraint. */
+  void BootstrapPRIs();
```

Comment:
Make private and call from `Bootstrap`

Reply:
Per previous conversation, needs builder refactor